controllers/fenceagentsremediation_controller.go

```diff
@@ -77,47 +79,68 @@ func (r *FenceAgentsRemediationReconciler) Reconcile(ctx context.Context, req ct
 	far := &v1alpha1.FenceAgentsRemediation{}
 	if err := r.Get(ctx, req.NamespacedName, far); err != nil {
 		if apiErrors.IsNotFound(err) {
-			// FAR is deleted, stop reconciling
-			r.Log.Info("FAR CR is deleted - nothing to do", "CR Name", req.Name, "CR Namespace", req.Namespace)
+			// Request object of FAR CR was not found, could have been deleted after reconcile request.
+			// Return and don't requeue
+			r.Log.Info("FenceAgentsRemediation CR was not found", "CR Name", req.Name, "CR Namespace", req.Namespace)
 			return emptyResult, nil
 		}
-		r.Log.Error(err, "failed to get FAR CR")
+		r.Log.Error(err, "Failed to get FenceAgentsRemediation CR")
 		return emptyResult, err
 	}
 	// Validate FAR CR name to match a nodeName from the cluster
 	r.Log.Info("Check FAR CR's name")
 	valid, err := utils.IsNodeNameValid(r.Client, req.Name)
 	if err != nil {
+		r.Log.Error(err, "Didn't find a node matching the CR's name", "CR's Name", req.Name)
 		return emptyResult, err
 	}
 	if !valid {
-		r.Log.Info("didn't find a node matching the CR's name", "CR's Name", req.Name)
+		r.Log.Error(err, "API notFound error for finding the node ", "CR's Name", req.Name)
 		return emptyResult, nil
 	}
 	// Fetch the FAR's pod
 	r.Log.Info("Fetch FAR's pod")
 	pod, err := utils.GetFenceAgentsRemediationPod(r.Client)
 	if err != nil {
+		r.Log.Error(err, "Can't find FAR's pod by it's label", "CR's Name", req.Name)
 		return emptyResult, err
 	}
 	//TODO: Check that FA is excutable? run cli.IsExecuteable
 
-	r.Log.Info("Create and execute the fence agent", "Fence Agent", far.Spec.Agent, "Node Name", req.Name)
+	// Build FA parameters
+	r.Log.Info("Combine fence agent parameters", "Fence Agent", far.Spec.Agent, "Node Name", req.Name)
 	faParams, err := buildFenceAgentParams(far)
 	if err != nil {
+		r.Log.Error(err, "Invalid sharedParameters/nodeParameters from CR", "CR's Name", req.Name)
 		return emptyResult, err
 	}
 	cmd := append([]string{far.Spec.Agent}, faParams...)
-	// The Fence Agent is excutable and the parameters are valid but we don't know about their values
-	if _, _, err := r.Executor.Execute(pod, cmd); err != nil {
-		//TODO: better seperation between errors from wrong shared parameters values and wrong node parameters values
+	// The Fence Agent is excutable and the parameters structure are valid, but we don't check their values
+	r.Log.Info("Execute the fence agent", "Fence Agent", far.Spec.Agent, "Node Name", req.Name)
+	outputRes, outputErr, err := r.Executor.Execute(pod, cmd)
+	if err != nil {
+		// response was a failure message
+		if outputErr != "" {
+			err = fmt.Errorf("%s- %w", outputErr, err)
+		}
+		r.Log.Error(err, "Fence Agent response was a failure", "CR's Name", req.Name)
+		return emptyResult, err
+	}
+	if outputRes != SuccessFAResponse {
```

Comment:
this looks risky to me. Are you sure that this will be the output by all fence-agent variations? I would expect that the command succeeded when err == nil && outputErr == ""

Reply:
Good catch as it would break the current E2E test since it uses status (and status has a different success message).
This logic suits better to https://github.com/medik8s/fence-agents-remediation/pull/54.

> Are you sure that this will be the output by all fence-agent variations?

I can verify it but I have got the impression that any FA from ClusterLabs has [specific actions](https://github.com/ClusterLabs/fence-agents/blob/main/doc/FenceAgentAPI.md#agent-operations-and-return-values) that it supports, and thus the output should be formatted the same. But I could be wrong, as I only tested on two FA so far.

> I would expect that the command succeeded when err == nil && outputErr == ""
 
I will use this term for now, but if my above impression/assumption is correct than it makes sense to restrict the the term that we check. In the future when we might want to support new and not FA from ClusterLabs, then we could ease this restriction...